docs/user/modeling.md

```diff
@@ -0,0 +1,144 @@
+
+| Type of field             | Field name                | Notes                                                                                                                                            |
+|---------------------------|---------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------|
+| Normal fields             | Has to match ORM exactly  | Make sure that the name matches the name in the ORM model.                                                                                       |
+| Custom fields             | Field name doesn't matter | Use `nautobot_ssot.contrib.CustomFieldAnnotation`                                                                                                |
+| Many-to-one relationships | Django lookup syntax      | See [here](https://docs.djangoproject.com/en/3.2/topics/db/queries/#lookups-that-span-relationships) - your model fields need to use this syntax |
+| One-to-many relationships | Has to match ORM exactly  | In case of a generic foreign key see [here](#special-case-generic-foreign-key)                                                                   | 
+
+
+## Normal Fields
+
+For normal, non-relationship, non-custom fields on a model, all you need to do is to ensure that field name on your SSoT model class matches that of the Nautobot model class. To ensure this is the case, you can check out the model either by reading the corresponding source code on [GitHub](https://github.com/nautobot/nautobot) or through the [Nautobot Shell](https://docs.nautobot.com/projects/core/en/stable/administration/nautobot-shell/) - in there you can use commands like `dir(Tenant)` to get an overview of available fields for the different models.
+
+!!! note
+    If you want to sync data into a model provided by a [Nautobot App](https://docs.nautobot.com/projects/core/en/stable/plugins/), you need to navigate to its respective source code repository. Furthermore, you may need to manually import the model if you're using the Nautobot Shell approach.
+
+## Custom Fields
+
+For [custom fields](https://docs.nautobot.com/projects/core/en/stable/models/extras/customfield/), you will need to use the `nautobot_ssot.contrib.CustomFieldAnnotation` class. Given a custom field called "Test" on the circuit provider model, this is how you could map that custom field into the diffsync model field `test_custom_field`:
+
+```python
+from typing_extensions import Annotated  # Python<3.9 - you need to install this module
+# from typing import Annotated  # Python>=3.9
+from nautobot.circuits.models import Provider
+from nautobot_ssot.contrib import NautobotModel, CustomFieldAnnotation
+
+class ProviderModel(NautobotModel):
+    _model = Provider
+    _modelname = "provider"
+    _identifiers = ("name",)
+    _attributes = ("custom_field",)
```

Comment:
to me, it's not clear the relationship between  "custom_field" and `test_custom_field`.

It maybe be that we only need to define one attribute "custom_field" and add add many different custom fields? or maybe, the magic comes from appending the slugify name of the custom field in the class field? I think that an example with two custom fields would deambiguate this question

Reply:
Actually, this is just a mistake. The attribute name still needs to match what's in `_attributes`